### PR TITLE
feat(awscdk-app-ts): add synth:silent cdk task

### PR DESCRIPTION
**Pull request by dannysteenman**
In a recent [pull request](https://github.com/projen/projen/pull/1323) the projen tasks like "projen synth" got updated and write their output to "/dev/null". 

I mostly run `projen synth` locally to debug my code by adding various console.log calls but now they get suppressed by default. So with this PR, I would like to introduce a new cdk task that adds surpression by running `projen synth:silent`. 

Another reason to revert the change is to stay consistent with the mapping of the task `cdk synth`. 

I've tested it in a new project and this is what the tasks look like in `.projen/tasks.json`:
```json
    "synth": {
      "name": "synth",
      "description": "Synthesizes your cdk app into cdk.out (part of \"yarn build\")",
      "steps": [
        {
          "exec": "cdk synth"
        }
      ]
    },
    "synth:silent": {
      "name": "synth:silent",
      "description": "Synthesizes your cdk app into cdk.out and surpresses the template in stdout (part of \"yarn build\")",
      "steps": [
        {
          "exec": "cdk synth > /dev/null"
        }
      ]
    },
```

and in `package.json`:
```json
    "synth": "npx projen synth",
    "synth:silent": "npx projen synth:silent",
```

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/awscdk/cdk-tasks.ts [hunk 1]
```diff
@@ -36,6 +41,11 @@ export class CdkTasks extends Component {
 
     this.synth = project.addTask('synth', {
       description: 'Synthesizes your cdk app into cdk.out (part of "yarn build")',
```
Comment (eladb) [inline src/awscdk/cdk-tasks.ts:38]: Remove "part of yarn build" here

**Review by eladb — CHANGES_REQUESTED** [review]
Where is the part that we switch which task is executed using "build"

File: src/awscdk/cdk-tasks.ts (hunk 1 above)
Comment (dannysteenman) [reply src/awscdk/cdk-tasks.ts:38]: Updated description and updated the postCompileTask in the build to use the silent synth cdk task

**Review by eladb — CHANGES_REQUESTED** [review]
Can you please add a unit test?

File: src/awscdk/awscdk-app-ts.ts [hunk 2]
```diff
@@ -91,7 +91,7 @@ export class AwsCdkTypeScriptApp extends TypeScriptAppProject {
     this.cdkTasks = new CdkTasks(this);
 
     // add synth to the build
-    this.postCompileTask.spawn(this.cdkTasks.synth);
+    this.postCompileTask.spawn(this.cdkTasks.synthSilent);
```
Comment (dannysteenman) [inline src/awscdk/awscdk-app-ts.ts:94]: Updated postCompileTask to use the silent synth cdk task.

**Review by eladb — APPROVED** [review]

**Comment by dannysteenman** [comment]
@eladb something is going wrong on the self mutation step in the build action. Is there something I need to enable/modify on my side to make it pass?

**Comment by eladb** [comment]
Yes, build locally and push any changes.

Self mutation doesn't work on forks (see https://github.com/projen/projen/issues/1362)

**Comment by dannysteenman** [comment]
somehow I missed the eslint correction, sorry for that.

**Pull request merged**